target/linux/ath79/dts/qca9557_ruckus_r500.dts

```diff
@@ -0,0 +1,324 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca955x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/mtd/partitions/uimage.h>
+
+/ {
+	compatible = "ruckus,r500", "qca,qca9557";
+	model = "Ruckus R500";
+
+	aliases {
+		led-boot = &power_green;
+		led-failsafe = &power_red;
+		led-running = &power_green;
+		led-upgrade = &power_red;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins &clks_disable_pins &enable_gpio_4>;
+
+		power_green: power-green {
+			function = LED_FUNCTION_POWER;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		power_red: power-red {
+			function = LED_FUNCTION_POWER;
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&gpio 16 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+			panic-indicator;
+		};
+
+		// DIR (Zone Director) LED - Indicates Zone director connection status
+		dir {
+			label = "green:dir";
```

Comment:
Why do you use `label` instead of the new format here and below?
Would be nice have leds sorted by their gpio.

Reply:
Upstream doesn't have a function defined, that would cover that, any idea what should be done in such cases?
Why sorted by GPIO and not grouped by LED function? In this case this would probably match, but I'd have to check that.

Reply:
> Upstream doesn't have a function defined, that would cover that, any idea what should be done in such cases?

https://github.com/openwrt/openwrt/pull/17114#discussion_r1874552259

> Why sorted by GPIO and not grouped by LED function?

That's most common practice. Reviewers and other people can easier check the dts.

Reply:
Okay, I switched other LEDs for color/function properties. I kept the previous `wlan2g` and `wlan5g` function names, which kernel doesn't define either - and this was precisely the reason I kept using 'label' here.
Regarding ordering of LEDs, I'd like to keep the current one for two reasons:
- it is consistent with other Ruckus devices I ported
- it matches the ordering on device front panel - I see this as more important, than ordering them by GPIOs, which isn't externally visible anywhere else besides `/sys/kernel/debug/gpio`.

Reply:
Hera are some examples how to deal with `wlan`:
https://github.com/openwrt/openwrt/blob/55b7b8f404fdbcfd865c379d34a61caaa1801755/target/linux/ramips/dts/mt7621_confiabits_mt7621-v1.dts#L74-L75

https://github.com/openwrt/openwrt/blob/55b7b8f404fdbcfd865c379d34a61caaa1801755/target/linux/ramips/dts/mt7621_confiabits_mt7621-v1.dts#L66-L67

Reply:
I applied Hauke's solution instead.

Reply:
This is better solution.